agent/http.go

```diff
@@ -262,6 +264,26 @@ func (s *HTTPServer) wrap(handler func(resp http.ResponseWriter, req *http.Reque
 	}
 }
 
+type handlerFunc func(resp http.ResponseWriter, req *http.Request) (interface{}, error)
+
+// requireAgentRead wraps the given function, requiring a token with agent read permissions
```

Comment:
Since we don't have other places in mind that will use this, I'd probably just make an `agent_endpoint.go:AgentMetrics()` method that has this logic, so it looks like the others and that can just call in to `s.agent.MemSink.DisplayMetrics` rather than wrapping. Can you add a unit test for ACLs as well?